### ISSUE TITLE
Implement Station Queue System

### ISSUE THREAD

**Issue by DragonmasterJ** [brata-hsdc/brata.masterserver#25]
This is just an enhancement idea I had when reviewing the messaging system. To help with organizing teams during the competition, the MS could manage a queue for station types based on the states of each station. 

We could leave this as information in our MS console to help crowd manage during the competition, or we can make it complicated and make it such that the teams are notified when a station is free and assigned a station programmatically.

**Comment by jawaad-ahmad** [comment]
Raised to low priority to match the rest; otherwise it'll get buried. We can re-evaluate priorities in a couple weeks after we get all the highs and mediums worked.

I like the idea of this. Given all that we would have to do, I don't see this getting worked till after March though.

Whenever we get around to this, I would envision this as something like having everyone sitting around as if in a waiting room, and each phone would get a toast or some other notification one-by-one as their turn comes up. This would free up folks to walk around while we monitor the size of each virtual "line" on the maintenance GUI or a dedicated area on the leader board.

One thing to keep in mind if we choose to do this would be what to do if a team is called and they don't show up to their station within a configurable period of time (e.g. 30-60 seconds). We don't want three stations to stay idle if their teams aren't showing up, so if the team doesn't scan the QR code within the allotted time, they get a follow-up notification that they lost their place in line and will have to wait again while another team gets notification that it's their turn now. No penalty.

The Master Server doesn't have to push them back to the end of the line again; it could insert them a configurable items down from the head.

**Comment by DragonmasterJ** [comment]
> One thing to keep in mind if we choose to do this would be what to do if a team is called and they don't show up to their station within a configurable period of time (e.g. 30-60 seconds). 

Yeah, I didn't even think about doing a timeout. That sounds awesome. 

> Whenever we get around to this, I would envision this as something like having everyone sitting around as if in a waiting room

I had the same idea, since we may want to add the animations for the docking and return to earth, if we have more displays going on in the main area of the room, teams can watch what is going on with the animations without having to worry about someone having to call them to stand in a queue, etc. 

This makes it even more engaging, almost enhancing Ellery's original idea to just have the animations in the first place.
